threatgrid/ctia
# Add asset entities to graphql route #4521

> **Epic** #
> Close 
> Related https://github.com/threatgrid/iroh/issues/4521

*Note: this is a multi-PR work. The second part is [here #1068 ](https://github.com/threatgrid/ctia/pull/1068)*

<a name="qa">[§](#qa)</a> QA
============================

1. Create one or multiple Assets by issuing requests to `POST ​/ctia​/asset` route
2. Try running simple graphql queries, e.g.: 

```
POST  /ctia/graphql
{
"query": "query Assets { assets { nodes {id} }}"
}
```

3. Make sure the results include all (previously created) assets
4. Similarly, test it for `AssetMapping`, `AssetProperties` and `TargetRecord`


<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: graphql queries for Asset, AssetMapping, AssetProperties and TargetRecord entities
public: one line description for public release notes.
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

2d651cc4 adds an asset-mapping specific test case
95357042 remove unused code
43710f3f adds target-record graphql tests
b97bc4f7 adds target-record graphql schema
6581d90c remove singular form of asset-properties
2b084878 adds AssetProperties tests
0ef440bf normalize ns headers
3785a4e9 adds asset-properties graphql schema
609fe710 adds basic asset-mapping graphql test
16bc4277 Adds graphql schemas for asset-mapping
214f34c1 extend asset graphql tests
883b3e3f move asset graphql schemas to its own namespace
657f7363 adds additional graphql schema things
7254460c adds AssetType to graphql/schemas
dd0f4d03 normalize and sort ns headers
